src/main.rs

```diff
@@ -333,52 +338,56 @@ mod tests {
 
     #[test]
     fn test_translate_atg() {
-        assert_eq!(translate("ATG", 1), 'M');
+        assert!(matches!(translate(b"ATG", 1), Ok('M')));
```

Comment:
Because `translate` now returns `Result<char, Error>`, I had to change the tests. [`matches!`](https://doc.rust-lang.org/std/macro.matches.html) is a macro equivalent to a `match` statement, but more ergonomic if you're checking for only one condition (in this case, `Ok('M')`). Since it returns `true` or `false`, the `assert!` will fail if the result is not the expected `'M'` value.